spec/features/teacher_sign_in_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe Teacher, type: :feature do
+  def sign_in(teacher)
+    fill_in 'teacher_email', with: teacher.email
+    fill_in 'teacher_password', with: teacher.password
+    click_on 'Log in'
+  end
+  sleep(3)
+  describe 'signs in' do
+    it 'successfully' do
+      Capybara.current_driver = :selenium_chrome
+      teacher = create(:teacher, attributes_for(:teacher))
+      visit root_path
+      click_link 'SignIn'
+      sign_in(teacher)
+      expect(page).to have_content 'Signed in successfully.'
+      sleep(6)
+    end
+
+    describe 'teacher is not signed_in' do
+      it 'with wrong credentials' do
+        Capybara.current_driver = :selenium_chrome
+        teacher = create(:teacher, attributes_for(:teacher))
+        visit root_path
+        wrong_credentials = build(:teacher,
```

Comment:
Since you're not using this `wrong_credentials` object to log in(there's no `sign_in` method invoked, and that's good), you can remove this line and later just use

`fill_in 'teacher_email', with: 'wrong_email@example.com'` . Less writing, less reading :)

Reply:
ping @mtrojanowska

Reply:
Missed this one somehow. Check :)